pkg/acsengine/template_generator.go

```diff
@@ -371,7 +371,11 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		},
 		"UseAksExtension": func() bool {
 			cloudSpecConfig := cs.GetCloudSpecConfig()
-			return cloudSpecConfig.CloudName == azurePublicCloud
+			return cloudSpecConfig.CloudName == api.AzurePublicCloud
+		},
+		"IsMooncake": func() bool {
+			cloudSpecConfig := cs.GetCloudSpecConfig()
+			return cloudSpecConfig.CloudName == api.AzureChinaCloud
```

Comment:
I would add a nil check here for `cloudSpecConfig`

Reply:
I don't see how `cloudSpecConfig` could ever be `nil`. The function returns a type struct (not a pointer to one), so even if the map lookup key didn't exist it would return a zero value `AzureEnvironmentSpecConfig` (which is not `nil`)